### PR TITLE
pr1

### PR DESCRIPTION
**Pull request by uckamdar59**
Added the driver api.

### CODE REVIEW

File: src/main/java/com/driver/service/DriverServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		
+		DriverResponse driverResponse = new DriverResponse();
+		Optional<Driver> d = driverRepository.findById(driverId);
+		if(d.isEmpty()) {
+			driverResponse.setStatus(Constants.AccountNotFoundError);
+			return driverResponse;
+		}
+		
+		driverResponse.setStatus(Constants.deleteSuccess);
+		driverRepository.delete(d.get());
```
Comment (badireddisatyasaivarun) [inline src/main/java/com/driver/service/DriverServiceImpl.java:141]: just inter-change code of line *140 and *141,
It is more appropriate as, after deletion, we assign the status.

File: src/main/java/com/driver/service/DriverServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+		return driverResponse;
```
Comment (badireddisatyasaivarun) [inline src/main/java/com/driver/service/DriverServiceImpl.java:142]: There is no use to return a response in delete when your function in controller is returning void.
We cannot be able to see the AccountNotFoundError too as controller function returns void

**Review by badireddisatyasaivarun — COMMENTED** [review]
correct these comments.

File: src/main/java/com/driver/service/DriverServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+			driverResponse.setStatus(Constants.phoneNumError1);
+			return driverResponse;
+		}else {
+			Matcher matcher = pattern.matcher(driverRequest.getPhoneNum());
+			if(!matcher.matches()) {
+				driverResponse.setStatus(Constants.phoneNumError2);
+				return driverResponse;
+			}
+				
+		}
```
Comment (badireddisatyasaivarun) [inline src/main/java/com/driver/service/DriverServiceImpl.java:69]: **under one transporterId** we can not post two **same phoneNumbers** in it, so before adding a phoneNumber check wether it is already added to the corresponding transporterId.

**Review by badireddisatyasaivarun — APPROVED** [review]

**Review by liveasy97 — APPROVED** [review]

**Pull request merged**